cmd/node/main.go

```diff
@@ -115,8 +117,22 @@ func main() {
 	if err != nil {
 		logger.Fatalf("fail to create kubernetes client, error: %v", err)
 	}
+	wrappedK8SClient := k8s.NewKubeClient(k8SClient, logger, *namespace)
+
+	k8sCache, err := k8s.GetK8SCache()
+	if err != nil {
+		logger.Fatalf("fail to create cache for kubernetes resources, error: %v", err)
+	}
+	// start cache
+	go func() {
+		// cache implementation we use newer returns err
+		_ = k8sCache.Start(stopCH)
```

Comment:
If caching go routine died will we start using non-cached mode, read outdated information or just fail?

Reply:
Cache implements retry logic. If WATCH call to kube API fails, reconnection will happen automatically. If there is no connection at the moment old data still can be readable from the cache.